packages/algorand-js/src/internal/interpreter/opcode-list.ts

```diff
@@ -0,0 +1,159 @@
+import { IStack } from "../../lib/stack";
+import { TealError } from "../core/errors";
+import { ERRORS } from "../core/errors-list";
+import type { AppArgs, StackElem } from "../types";
+import { parseNum } from "../utils/bigint";
+import { Op } from "./opcode";
+
+const BIGINT0 = BigInt("0");
+// pops string([]byte) from stack and pushes it's length to stack
+export class Len extends Op {
+  execute (stack: IStack<StackElem>): void {
+    this.assertStackLen(stack, 1);
+    const top = stack.pop();
+    if (typeof top !== 'string') {
+      throw new TealError(ERRORS.TEAL.INVALID_OP_ARG, {
+        opcode: "len"
+      });
+    }
+    stack.push(BigInt(top.length));
+  }
+}
+
+// pops two unit64 from stack(a, b) and pushes their sum(a + b) to stack
+// panics on overflow (result > max_unit64)
+export class Add extends Op {
+  execute (stack: IStack<StackElem>): void {
+    this.assertStackLen(stack, 2);
+    const a = stack.pop();
+    const b = stack.pop();
+    if (typeof a !== 'bigint' || typeof b !== 'bigint') {
+      throw new TealError(ERRORS.TEAL.INVALID_OP_ARG, {
+        opcode: "+"
+      });
+    }
```

Comment:
let's create a function to handle this:
```
function assertBigInt(a: unknown)
```

Reply:
And let's use that function in all operators below:
```
assertBigInt(a);
assertBigInt(b);
```

Reply:
ok